portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -5,23 +5,52 @@
 import com.google.cloud.datastore.Entity;
 import com.google.cloud.datastore.FullEntity;
 import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.text.StringEscapeUtils;
 
 @WebServlet("/form-handler")
 public class FormHandlerServlet extends HttpServlet {
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    Query<Entity> query =
+        Query.newEntityQueryBuilder().setKind("contact").setOrderBy(OrderBy.desc("name")).build();
+    QueryResults<Entity> results = datastore.run(query);
+
+    List<String> contacts = new ArrayList<>();
```

Comment:
Move this block also to a separate method. Something like getContacts() will make it clear what it does.

The method can read from datastore and then build the contact list
List<String> getContacts() {
   QueryResult<Entity> results = readFromDatastore();
    // Block of code to extract contacts.
}